bee-node/bee-node/src/fullnode/builder.rs

```diff
@@ -207,14 +208,10 @@ fn add_node_resources<S: NodeStorageBackend>(builder: FullNodeBuilder<S>) -> Res
     // TODO block ? Make new async ?
     let storage = S::start(storage_cfg).map_err(|e| CoreError::StorageBackend(Box::new(e)))?;
 
-    if config.local().is_new() {
+    if !config.identity().is_restored() {
         storage
             .set_health(StorageHealth::Healthy)
             .map_err(|e| CoreError::StorageBackend(Box::new(e)))?;
-
-        return Err(FullNodeError::InvalidOrNoIdentityPrivateKey(
```

Comment:
Why is this removed ?

Reply:
Because ... if the identity hasn't been restored ... then a new valid one was created.

Reply:
Actually I am not sure I still understand the logic behind that if-branch. Why are we setting the storage to healthy and then exit?

Reply:
Clarified that the whole if-branch can be removed.